Description: nall/Makefile fixes

* Use linux platform settings also on kfreebsd.
* Use the flags from CFLAGS, CPPFLAGS and LDFLAGS. They are set in the rules file.

Author: Tobias Hansen <thansen@debian.org>

--- a/nall/GNUmakefile
+++ b/nall/GNUmakefile
@@ -30,7 +30,7 @@
   else ifneq ($(findstring Linux,$(uname)),)
     platform := linux
   else ifneq ($(findstring BSD,$(uname)),)
-    platform := bsd
+    platform := linux
   else
     $(error unknown platform, please specify manually.)
   endif
@@ -223,7 +223,8 @@
 
 # linux settings
 ifeq ($(platform),linux)
-  options += -ldl
+  flags += $(CFLAGS) $(CPPFLAGS)
+  options += -ldl $(LDFLAGS)
 endif
 
 # bsd settings
